packages/bot-web-ui/src/components/dashboard/dashboard-component/load-bot-preview/recent-workspace.tsx

```diff
@@ -63,46 +66,65 @@ const RecentWorkspace = observer(({ workspace, index }: TRecentWorkspace) => {
         setSelectedStrategyId(workspace.id);
     };
 
+    const handleInit = () => {
+        setPreviewedStrategyId(workspace?.id);
+        if (active_tab === 0) {
+            previewRecentStrategy(workspace.id);
+        }
+    };
+
+    const handlePreviewList = () => {
+        setPreviewedStrategyId(workspace.id);
+        if (is_mobile) {
+            setPreviewOnDialog(true);
+            const dashboard_tab_dom_element = document.getElementsByClassName('tab__dashboard')?.[0];
+            waitForDomElement('#load-strategy__blockly-container', dashboard_tab_dom_element).then(() => {
+                previewRecentStrategy(workspace.id);
+            });
+        }
+    };
+
+    const handleEdit = async () => {
+        await loadFileFromRecent();
+        setActiveTab(DBOT_TABS.BOT_BUILDER);
+    };
+
+    const handleSave = () => {
+        updateBotName(workspace?.name);
+        toggleSaveModal();
+    };
+
```

Comment:
@farabi-deriv I would suggest you consider the movement of these functions to the store (probably load_modal) since we have to pass them inner functions to `RecentWorkspace component `anyway
the part of the code:
```
    const {
        dashboard_strategies,
        getRecentFileIcon,
        getSaveType,
        getSelectedStrategyID,
        loadFileFromRecent,
        onToggleDeleteDialog,
        previewRecentStrategy,
        previewed_strategy_id,
        selected_strategy_id,
        setSelectedStrategyId,
        setPreviewedStrategyId,
    } = load_modal;
```

Reply:
@maryia-matskevich-deriv I don't think that would be wise as we are trying to depend less on stores going forward. Also the handler functions here are component specific, so not sure why we should put this in the store instead of putting in this component.

Reply:
@farabi-deriv The point is **not** to pull 10 functions from the store and then use them inside 4 functions, but to pull **only 4** functions from the store.🙂

Reply:
it's a kind of optimization. You've already started 10 functions in the store and now you're throwing them into the component, what for? And you can throw the function you use in the component into the component. There are situations when you don't need to start functions in the store, but that's another case

Reply:
but it's good in case all the functions are in one store

Reply:
@sandeep-deriv @rupato-deriv what do you think guys about it?

Reply:
@maryia-matskevich-deriv I understand your point but the 4 functions that you are asking to create inside the store are only gonna be used in this component and not anywhere else. So thats why i don't see the point of putting them in store. Since its not shared by any other component. Adding those 4 functions in the store will make the store even more heavy. And our stores are already cluttered and the plan is to reduce this.

Reply:
@farabi-deriv this answer already sounds more convincing 😊 
Since most of the functions written in storages are most often used in one particular component. So now we have to pull all such functions into components?🤨
Here is an example of several types of functions that are better placed in components:
1. Display Functions (Render Logic)
2. Event Handlers
3. Lifecycle Methods(hooks)
4. Local State
5. Side Effects

Reply:
@maryia-matskevich-deriv Yes we might have to do that eventually as there are already talks of using hooks instead of stores. Thats why i was just trying to tell you that atleast for now going forward lets not put it in store.

Reply:
For the reference of what I'm talking about the use of store going forward https://deriv-group.slack.com/archives/C3AHMSM3P/p1691581217545619